src/modules/dashboard/components/ColonyActions/ColonyActions.tsx

```diff
@@ -91,6 +102,23 @@ const ColonyActions = ({
     },
   });
 
+  /* filtering at this level as it is more performant
+  & we reduce the number of events passing through transformers */
+  const uniqueEvents = useMemo(
+    () =>
+      /* additional check for types to work */
+      data === undefined
+        ? []
+        : data.events.filter(
+            (action) =>
+              !data?.oneTxPayments.some(
+                (paymentAction) =>
+                  paymentAction.transaction?.hash === action.transaction.hash,
+              ),
+          ),
+    [data],
```

Comment:
This whole thing needs a try/catch block.

We have colonies (broken ones) right now on QA that don't have `action.transaction`

Yes they are broken, but just because that it doesn't mean we can't provision and provide some level of fallback for that.

Reply:
I really don't like how this is so hard to read, but due to time constraints it will have to do